### PR TITLE
Enh/transmit unencoded

### PR DESCRIPTION
**Pull request by BeckerMax**
Accepts the flag `Transmit Unencoded`.

Needed for creation of bootstrap-tokens for machines introduced here: https://github.com/gardener/gardener/pull/3902

**How to categorize this PR?**
/area security
/kind enhancement
/os garden-linux

**What this PR does / why we need it**:
See https://github.com/gardener/gardener/issues/3898 and https://github.com/gardener/gardener/pull/3902

**Release note**:
```other operator
Introduces new flag `TransmitUnencoded` which writes file content unencoded into the worker resource.
```

### CODE REVIEW

**Comment by BeckerMax** [comment]
If you want to test this PR here is a image with this version:
eu.gcr.io/gardener-project/gardener/extensions/os-gardenlinux:v0.9.0-dev-mb.3
Remember to also update the helm chart in the controllerregistration (helm chart changed independently of this PR).

**Comment by rfranzke** [comment]
#3902 got merged, please revendor it and mark this PR as 'ready for review'

File: pkg/generator/templates/cloud-init.gardenlinux.template [hunk 1]
```diff
@@ -6,6 +6,12 @@ cat << EOF | base64 -d > '{{ .Path }}'
 EOF
 {{- end -}}
 
+{{- define "put-content-unencoded" -}}
```
Comment (vpnachev) [inline pkg/generator/templates/cloud-init.gardenlinux.template:9]:
```suggestion
{{- define "put-content-decoded" -}}
```

or

```suggestion
{{- define "put-content-raw" -}}
```

Or even  
```suggestion
{{- define "put-content" -}}
```

and rename the  the current `put-content` to `put-content-encoded`.

Comment (BeckerMax) [reply pkg/generator/templates/cloud-init.gardenlinux.template]: Default behaviour is to place it encoded. Special case is `raw`. So I now use `put-content` for the default case and `put-content-raw` for the special case. Thanks for the suggestion.

**Review by rfranzke — APPROVED** [review]
/lgtm

**Comment by BeckerMax** [comment]
I guess we wait before cutting a new version of the os-extension provider until we released a new version of g/g ? So the go.mod points to a proper version and not a commit?

**Review by vpnachev — APPROVED** [review]
/lgtm

> I guess we wait before cutting a new version of the os-extension provider until we released a new version of g/g ? So the go.mod points to a proper version and not a commit?

I do not see a problem when a commit is used.

**Comment by rfranzke** [comment]
@BeckerMax Please open similar PRs for the other OS extensions.

**Pull request merged**